backbone.js

```diff
@@ -38,6 +38,13 @@
   var array = [];
   var slice = array.slice;
 
+  // Optimized deep comparison function.
+  var _isEqual = _.isEqual;
+
+  function isEqual( a, b ){
+      return a && b && typeof a === 'object' && typeof b === 'object' ? _isEqual( a, b ) : a === b;
```

Comment:
_.isEqual is terribly slow on typical Model.set use cases, like assignments of primitive values like integers. This is not just "faster", but in fact it's 10x faster (open profiler and see), no matter what someone might expect. No one can really predict how JIT works on real program.